### ISSUE TITLE
Tiling Artefact (plant-seg 2.0.0a5)

### ISSUE THREAD

**Issue by Amerebeing** [kreshuklab/plant-seg#367]
Getting tiling artefact when using napari plugin of plantseg. I am using lightsheet_3D_unet_root_nuclei_ds1x model to segment my nuclei.  The tiling artefact is affecting the final segmentation. 
![image](https://github.com/user-attachments/assets/01153e85-80a4-48a1-8ef1-49ad2e58cf56)

**Comment by qin-yu** [comment]
Hey @Amerebeing thank you for reporting this issue!

I offer two things to try - either you send me this volume for me to try it; or you can try these yourself:

1. rescale your volume to match the voxel size of the model by "Set Voxel Size" and "Rescale Image (to model voxel size)";
2. increase the halo from 44 to, say, 64, in "Show advanced parameters" in "Image to Prediction", but you may need to reduce the patch shape accordingly.
3. just try `PlantSeg_3Dnuc_Platinum`

You can do these separately or together and see if the prediction is improved.

**Comment by qin-yu** [comment]
I just tried all three separately on one of my dateset, and all improved.

**Comment by qin-yu** [comment]
This may relate to:

- #224

**Comment by Giannisbiol** [comment]
Hi,

I get also these kind of "thin" labels. I have noticed that playing a bit with the watershed and the agglomeration method, these can totally (mostly) vanish or become more abundant. If the boundary prediction is perfect though, then it looks even better. Unfortunately, in my case (roots), there is no standard set up  of the segmentation. Most of the time if it is good on one image, it will be ok on the next one too but it can  vary, even if acquisition parameters are all the same.

**Comment by qin-yu** [comment]
Did you try the `PlantSeg_3Dnuc_platinum` model? It gives a nuclei probability map and a nuclei boundary probability map. If you use the [Remove Objects with Low Foreground Probability](https://kreshuklab.github.io/plant-seg/chapters/plantseg_interactive_napari/proofreading_auto/#widget-use-foreground-to-remove-false-positives) widget, you can remove the false positives.

**Comment by qin-yu** [comment]
I just tried to run `PlantSeg_3Dnuc_platinum` and default 3D watershed + GASP on your image and the result look okay: 

```
segmentation  | raw
boundary pred | foreground pred
```

![image](https://github.com/user-attachments/assets/679fde36-d9b2-4e34-ae38-a0547cc060ce)

**Comment by qin-yu** [comment]
Hey @Amerebeing, [GoNuclear repo](https://github.com/kreshuklab/go-nuclear/tree/main) has three models: Cellpose, PlantSeg and StarDist, where `PlantSeg_3Dnuc_platinum` is the PlantSeg model. Here I used GoNuclear's `run-stardist` for you: [Side_rAP2_16LDs_SAM4_nuclei_Z0.400_X0.291_Y0.291_Sz453_Sx800_Sy800_segmentation.tif.zip](https://github.com/user-attachments/files/18008976/Side_rAP2_16LDs_SAM4_nuclei_Z0.400_X0.291_Y0.291_Sz453_Sx800_Sy800_segmentation.tif.zip). If you are interested, check how to use it here: https://github.com/kreshuklab/go-nuclear/tree/main/stardist

![image](https://github.com/user-attachments/assets/e0349f5d-18b0-4452-abfc-30d8f5581390)

**Comment by Amerebeing** [comment]
hey @qin-yu The Platinum model performs well on my 3D volume when I downsample it by a factor of 2. Adjusting the Watershed and GASP parameters also helps in reducing artifacts in the final segmentation. I’d like to know the original voxel size the Platinum model was trained on, as my original volume (voxel size: 0.30 × 0.17 × 0.17) produces a faded probability map without downsampling.
![image](https://github.com/user-attachments/assets/7707b08e-32b3-4647-a2e9-ca91639e5623)

On downsampling i am getting better results 
![image](https://github.com/user-attachments/assets/0f626701-e982-4eeb-8403-70d83b7b520e)

**Comment by qin-yu** [comment]
Hey @Amerebeing, for your convenience, here’s the voxel size:

```python
original_voxel_size = {  # z, y, x
    1135: [0.28371836501901143, 0.12678642066720086, 0.12678642066720086],  # validation
    1136: [0.2837183895131086,  0.12756971653115998, 0.12756971653115998],  # training
    1137: [0.2837183895131086,  0.1266211463645486,  0.1266211463645486 ],  # training
    1139: [0.2799036917562724,  0.12674335484590543, 0.12674335484590543],  # training
    1170: [0.27799632231404964, 0.12698523961670266, 0.12698522349145364],  # training
}  # [0.2837, 0.1268, 0.1268] is taken as the median

original_median_extents = {  # z, y, x
    1135: [16, 32, 33],  # validation
    1136: [16, 32, 32],  # training
    1137: [16, 32, 32],  # training
    1139: [16, 32, 33],  # training
    1170: [16, 29, 30],  # training
    'average':
}  # [16, 32, 32] is taken as the median
```

For reference, the training data for my Platinum model is described in the GoNuclear repository: [GoNuclear Training Data](https://github.com/kreshuklab/go-nuclear?tab=readme-ov-file#training-data). You can also find detailed information in the paper: [Materials and Methods](https://journals.biologists.com/dev/article/151/14/dev202800/361191/A-deep-learning-based-toolkit-for-3D-nuclei).

The training data itself is available at the BioImage Archive: [S-BIAD1026](https://www.ebi.ac.uk/biostudies/BioImages/studies/S-BIAD1026?query=%20S-BIAD1026).

**Comment by qin-yu** [comment]
Hey @Amerebeing is the issue solved?

**Comment by Amerebeing** [comment]
Hey @qin-yu, thanks for checking in! I scaled my 3D volume by a factor of 0.7, and it's working perfectly. Any scaling below 0.7 causes the nuclei to merge, but the range of 0.7 to 0.75 seems to be the sweet spot. I'm quite happy 😊 with the results!
